src/external/clinical/types.ts

```diff
@@ -17,101 +17,106 @@
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+import { z as zod } from 'zod';
+
+//  Record<string, string | number | boolean | undefined>;
+export const ClinicalInfo = zod.record(zod.union([zod.string(), zod.number(), zod.boolean()]).optional());
+
+export const ClinicalSample = zod.object({
+  clinicalInfo: ClinicalInfo,
+  sampleId: zod.string(),
+  submitterId: zod.string(),
+  sampleType: zod.string(),
+});
+export const ClinicalSpecimen = zod.object({
+  clinicalInfo: ClinicalInfo,
+  samples: zod.array(ClinicalSample),
+  specimenId: zod.string(),
+  submitterId: zod.string(),
+  tumourNormalDesignation: zod.string(),
+  specimenType: zod.string(),
+  specimenTissueSource: zod.string(),
+});
+// TODO: The properties we want to index from the following types should be declared.
+export const ClinicalFollowUp = zod.object({
+  clinicalInfo: ClinicalInfo,
+  followUpId: zod.string(),
+});
+export const ClinicalPrimaryDiagnosis = zod.object({
+  clinicalInfo: ClinicalInfo,
+  primaryDiagnosisId: zod.string(),
+});
+export const ClinicalTherapy = zod.object({
+  clinicalInfo: ClinicalInfo,
+  therapyId: zod.string(),
+});
+export const ClinicalTreatment = zod.object({
+  clinicalInfo: ClinicalInfo,
+  therapies: zod.array(ClinicalTherapy),
+  treatmentId: zod.string(),
+});
+export const ClinicalFamilyHistory = zod.object({
+  clinicalInfo: ClinicalInfo,
+  familyHistoryId: zod.string(),
+});
+export const ClinicalExposure = zod.object({
+  clinicalInfo: ClinicalInfo,
+  exposureId: zod.string(),
+});
+export const ClinicalComorbidity = zod.object({
+  clinicalInfo: ClinicalInfo,
+  comorbidityId: zod.string(),
+});
+export const ClinicalBiomarker = zod.object({
+  clinicalInfo: ClinicalInfo,
+  biomarkerId: zod.string(),
+});
+
 /**
  * The specific content in ClinicalDonor type is subject to change as the data-dictionary is updated,
  * so we try to only define here the bare minimum of fields that are needed to interact with and index the clinical data.
+ * For this reason, the Zod schema must allow additional properties to be present (not strict).
  *
  * Note: we receive all dates as strings. The type declaration here has them listed as strings, but if it makes sense in the future
- * we can parse them into proper Date objects when the `JSON.parse(donor)` is called.
+ * we can update the Zod schema to parse them into dates.
  */
-export type ClinicalDonor = {
-  donorId: string;
-  gender: string;
-  programId: string;
+export const ClinicalDonor = zod.object({
```

Comment:
Convert type to Zod to get request validation. We rely on the `completionStats` section.